pkg/apis/seedmanagement/v1alpha1/defaults_managedseed.go

```diff
@@ -116,7 +116,19 @@ func setDefaultsGardenletConfiguration(obj *gardenletv1alpha1.GardenletConfigura
 
 	// Initialize seed config
 	if obj.SeedConfig == nil {
-		obj.SeedConfig = &gardenletv1alpha1.SeedConfig{}
+		obj.SeedConfig = &gardenletv1alpha1.SeedConfig{
+			SeedTemplate: gardencorev1beta1.SeedTemplate{
+				Spec: gardencorev1beta1.SeedSpec{},
+			},
+		}
+	}
+
+	if obj.SeedConfig.SeedTemplate.Spec.Ingress == nil {
+		obj.SeedConfig.SeedTemplate.Spec.Ingress = &gardencorev1beta1.Ingress{
+			Controller: gardencorev1beta1.IngressController{
+				Kind: "nginx",
+			},
+		}
```

Comment:
I am not sure if this defaulting belongs here. The `ingress.domain` field is being defaulted/initialized in the validator admission plugin: https://github.com/gardener/gardener/blob/024b9f343303febe4a8def668dc55f6eb78b812b/plugin/pkg/managedseed/validator/admission.go#L371-L376

Defaulting here only the ingress controller kind and the domain to be defaulted somewhere else does not make sense. I guess the ingress kind also have to be defaulted in the admission plugin? WDYT?

Reply:
OK, when thinking about it again let's keep it as it is.

Reply:
In terms of code locality your suggestion makes sense. But conceptually what is the purpose of validator admission plugin? Why is there defaulting logic in the first place? What is the context in which you want to disable it and the defaulting in it not to happen?

Reply:
> But conceptually what is the purpose of validator admission plugin?

In admission plugins we usually do validations (the cross-object validation, not the static one) and defaulting. The defaulting is cross-object in our particular case - we have to fetch the Shoot resource to compute the ManagedSeed's ingress domain and later on to validate it. Historically, we don't do this operation in the static validation or defaulting for the ManagedSeed as there referenced resource is not available there.

> Why is there defaulting logic in the first place?

I am also not very happing that admission plugins with validator in their name are doing defaulting as well. Years ago I created https://github.com/gardener/gardener/issues/2158 but I guess no one had capacity to work on it.

> What is the context in which you want to disable it and the defaulting in it not to happen?

I don't have any productive use case in might. You see that in tests this is the case. And theoretically you are allowed to do it. 

---

Maybe indeed I would move the defaulting to the admission plugin because:
- the current logic for setting the ingress field in the admission plugin is incomplete. It requires right now something or someone to specify the controller.kind field
- the current defaulting in this PR is incomplete as it requires something or someone to specify the ingress.domain field

WDYT?

Reply:
Your suggestion will increase code locality so I agree with it